aten/src/ATen/core/jit_type.h

```diff
@@ -294,58 +265,67 @@ inline c10::optional<T> merge_primitive(
 // an empty c10::optional in `sizes_`. If a rank is dynamic, the entire
 // `sizes_` becomes the empty optional.
 struct CAFFE2_API VaryingShape {
+  using ListOfOptionalInts = std::vector<c10::optional<int64_t>>;
   VaryingShape(const std::vector<int64_t>& vec)
-      : size_(vec.size()), dims_(vec.begin(), vec.end()) {}
+      : VaryingShape(ListOfOptionalInts(vec.begin(), vec.end())) {}
```

Comment:
Errr, deja vu, isn't there another PR floating with this exact change?

Reply:
yeah, I split that out from this one, because it is higher priority to merge that one to prevent a hang in some client code. I will delete these changes from this PR.